### PR TITLE
Changes in "DiffIn..." functions and also testcase fix

### PR DESCRIPTION
**Pull request by ItsRanveer**
DiffInMonth and DiffInYear were returning 1 for different month and years even if the interval was less than 1 hour.

### CODE REVIEW

File: carbon.go [hunk 1]
```diff
@@ -1035,21 +1047,44 @@ func (c *Carbon) DiffInYears(carb *Carbon, abs bool) int64 {
...
+	diffHr := c.DiffInHours(carb, abs)
+	hrLastYear := int64(c.DaysInYear() * hoursPerDay)
+
+	if (diffHr - hrLastYear) >= 0 {
+		diff := int64(carb.In(time.UTC).Year() - c.In(time.UTC).Year())
+
+		return absValue(abs, diff)
+	}
+
+	return absValue(abs, 0)
```
Comment (hfcorreia) [inline carbon.go]: You can immediately return 0 in this case

File: carbon.go [hunk 2]
```diff
@@ -1035,21 +1047,44 @@ func (c *Carbon) DiffInYears(carb *Carbon, abs bool) int64 {
...
+	diffHr := c.DiffInHours(carb, abs)
+	hrLastMonth := int64(c.DaysInMonth() * hoursPerDay)
+
+	if (diffHr - hrLastMonth) >= 0 {
+		diff := c.DiffInYears(carb, abs)*monthsPerYear + int64(carb.In(time.UTC).Month()-c.In(time.UTC).Month())
+
+		return absValue(abs, diff)
+	}
+
+	return absValue(abs, 0)
```
Comment (hfcorreia) [inline carbon.go]: return 0

File: carbon.go [hunk 3]
```diff
@@ -1035,21 +1047,44 @@ func (c *Carbon) DiffInYears(carb *Carbon, abs bool) int64 {
...
+}
+
+// DiffInString return the difference between duration in string format
+func (c *Carbon) DiffInString(carb *Carbon) string {
+	if carb == nil {
+		carb = nowIn(c.Location())
+	}
+
+	return strings.Replace(carb.Sub(c.Time).String(), "-", "", 1)
```
Comment (hfcorreia) [inline carbon.go]: This should diff the dates and then format it using the current `stringFormat`.

**Comment by hfcorreia** [comment]
You could add a test case for the `DiffInString(nil)`

Reviewed
